### PR TITLE
feat: added endpoint to get availability for a specific doctor.

### PR DESCRIPTION
**Pull request by Kavindu-248**

### CODE REVIEW

File: apps/assesment/views.py [hunk 1]
```diff
@@ -28,6 +28,11 @@ def get_queryset(self):
         if self.request.user.role == Roles.PATIENT:
             return Avalability.objects.filter(patient=self.request.user.patient, booked=False)
 
+    @action(detail=False, methods=['get'], url_path='(?P<doctor_id>[^/.]+)/availabilities')
```
Comment (Raam124) [inline apps/assesment/views.py:31]: first, get the doctor and check if the doctor exists or throw 404. Then get the availabilities using the related name

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 1 above)
Comment (Kavindu-248) [reply apps/assesment/views.py]: Change made

File: apps/assesment/views.py [hunk 2]
```diff
@@ -28,6 +29,13 @@ def get_queryset(self):
         if self.request.user.role == Roles.PATIENT:
             return Avalability.objects.filter(patient=self.request.user.patient, booked=False)
 
+    @action(detail=False, methods=['get'], url_path='(?P<doctor_id>[^/.]+)/avalabilities')
+    def get_availability_for_doctor(self, request, doctor_id):
+        doctor = get_object_or_404(Doctor, pk=doctor_id)
+        queryset = Avalability.objects.filter(doctor=doctor, booked=False)
```
Comment (Raam124) [inline apps/assesment/views.py:35]: use related names here

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 2 above)
Comment (Kavindu-248) [reply apps/assesment/views.py]: Change added

File: apps/assesment/views.py [hunk 3]
```diff
@@ -28,6 +29,13 @@ def get_queryset(self):
         if self.request.user.role == Roles.PATIENT:
             return Avalability.objects.filter(patient=self.request.user.patient, booked=False)
 
+    @action(detail=False, methods=['get'], url_path='(?P<doctor_id>[^/.]+)/avalabilities')
+    def get_availability_for_doctor(self, request, doctor_id):
+        get_object_or_404(Doctor, pk=doctor_id)
+        queryset = Avalability.objects.filter(doctor__id=doctor_id, booked=False )
```
Comment (Raam124) [inline apps/assesment/views.py:35]: not like this use the related name to fetch the related objects refer the docs

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/assesment/views.py (hunk 3 above)
Comment (Kavindu-248) [reply apps/assesment/views.py]: Amended the code

**Review by Raam124 — APPROVED** [review]

**Pull request merged**